Source/Plugin.BLE/Android/BroadcastReceivers/BondStatusBroadcastReceiver.cs

```diff
@@ -1,41 +1,59 @@
 using System;
 using Android.Bluetooth;
 using Android.Content;
-using Plugin.BLE.Abstractions;
+using Android.OS;
 using Plugin.BLE.Abstractions.EventArgs;
 using Plugin.BLE.Android;
+using Plugin.BLE.Extensions;
 
 namespace Plugin.BLE.BroadcastReceivers
 {
-    //[BroadcastReceiver]
     public class BondStatusBroadcastReceiver : BroadcastReceiver
     {
+	    private readonly Adapter _broadcastAdapter;
+
         public event EventHandler<DeviceBondStateChangedEventArgs> BondStateChanged;
 
+        public BondStatusBroadcastReceiver(Adapter adapter)
+        {
+            _broadcastAdapter = adapter;
+        }
+
         public override void OnReceive(Context context, Intent intent)
         {
-            var bondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
-            //ToDo
-            var device = new Device(null, (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice), null);
-            Console.WriteLine(bondState.ToString());
+            if (BondStateChanged == null)
+            {
+	            return;
+            }
 
-            if (BondStateChanged == null) return;
+            var extraBondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
 
-            switch (bondState)
+            BluetoothDevice bluetoothDevice;
+            
+            // As older versions of .NET (pre7) don't include the Tiramisu version code, handle it manually.
+#if NET7_0_OR_GREATER
+	        if (Build.VERSION.SdkInt >= BuildVersionCodes.Tiramisu)
+#else
+            if ((int)Build.VERSION.SdkInt >= 33)
+#endif
```

Comment:
One comment here: The usual pattern for this kind of SDK version check in other parts of the library is ...

```
#if NET6_0_OR_GREATER
                    OperatingSystem.IsAndroidVersionAtLeast(33)
#else
                    (Build.VERSION.SdkInt >= BuildVersionCodes.Tiramisu)
-#endif
```

Is there a particular reason why you chose to do it differently, or just "trying to make CI work somehow"? If I understand your comment correctly, then the `Tiramisu` version code is not available everywhere (although I don't fully understand why). Maybe just use the number instead, but stick to the rest of the pattern?

I think the idea of the above pattern was to use the modern way of doing this on recent .NET versions (i.e. `OperatingSystem.IsAndroidVersionAtLeast`), and the legacy way on .NET Standard / Xamarin (i.e. `Build.VERSION.SdkInt`).

I would prefer to make all such checks in a consistent manner, unless there is a good reason to deviate from it.

Reply:
I changed it to your implementation, yet again that throws the error that Tiramisu is not available, see the failed github action:  
`D:\a\dotnet-bluetooth-le\dotnet-bluetooth-le\Source\Plugin.BLE\Android\BroadcastReceivers\BondStatusBroadcastReceiver.cs(37,59): error CS0117: 'BuildVersionCodes' does not contain a definition for 'Tiramisu'`  
Is a cast alright in this case? 

I didn't know that `OperatingSystem.IsAndroidVersionAtLeast(xyz)` is the preferred way in newer versions, updated it accordingly.

Reply:
> I changed it to your implementation, yet again that throws the error that Tiramisu is not available (which I tried to evade by my implementation), see the failed github action: `D:\a\dotnet-bluetooth-le\dotnet-bluetooth-le\Source\Plugin.BLE\Android\BroadcastReceivers\BondStatusBroadcastReceiver.cs(37,59): error CS0117: 'BuildVersionCodes' does not contain a definition for 'Tiramisu'` Is a cast alright in this case?

I think the failure is due to the fact that we're not targeting Android 13 yet. So, yes, casting to int is ok with me. Thanks for your adjustments!